modules/components/tag-input/tag-input.ts

```diff
@@ -929,7 +917,7 @@ export class TagInputComponent extends TagInputAccessor implements OnInit, After
             .of(this.getItemDisplay(item))
             .map(display => this.setInputValue(display))
             .filter(display => this.inputForm.form.valid && !!display)
-            .map((display: string) => this.createTag(fromAutocomplete ? item : display))
+            .map((display: string) => this.createTag(item))
```

Comment:
as discussed - the items needs to undergo processing before it can be appended, which is why it needs to be display, rather than item, unless it comes from the autocomplete